src/compiler/scala/tools/nsc/tasty/TastyReader.scala

```diff
@@ -0,0 +1,142 @@
+package scala.tools.nsc.tasty
+
+import TastyRefs._
+import collection.mutable
+
+/** A byte array buffer that can be filled with bytes or natural numbers in TASTY format,
+ *  and that supports reading and patching addresses represented as natural numbers.
+ *
+ *  @param bytes    The array containing data
+ *  @param start    The position from which to read
+ *  @param end      The position one greater than the last byte to be read
+ *  @param base     The index referenced by the logical zero address Addr(0)
+ */
+class TastyReader(val bytes: Array[Byte], start: Int, end: Int, val base: Int = 0) {
```

Comment:
These classes are duplicated from the `tasty-core` project in dotty https://github.com/lampepfl/dotty/tree/master/tasty/src/dotty/tools/tasty I hoped we could depend on it, but its compiled with scala 2.13.1, so probably not allowed